src/providers/FileSystemProvider/FileSearchProvider.ts

```diff
@@ -44,13 +49,17 @@ export class FileSearchProvider implements vscode.FileSearchProvider {
     // When this is called without a query.pattern, every file is supposed to be returned, so do not provide a filter
     let filter = "";
     if (pattern.length) {
-      pattern = !csp ? query.pattern.replace(/\//g, ".") : query.pattern;
+      let escapeClause = "";
+      pattern = !csp ? pattern.replace(/\//g, ".") : pattern;
       if (pattern.includes("_") || pattern.includes("%")) {
         // Need to escape any % or _ characters
-        filter = `Name LIKE '%${pattern.replace(/(_|%|\\)/g, "\\$1")}%' ESCAPE '\\'`;
-      } else {
-        filter = `Name LIKE '%${pattern}%'`;
+        pattern = pattern.replace(/(_|%|\\)/g, "\\$1");
```

Comment:
Why not always do this replacement and provide the ESCAPE clause? I doubt it impacts performance significantly and it makes this code shorter:

```ts
filter = `Name LIKE '%${pattern.replace(/(_|%|\\)/g, "\\$1").replace(/\*/g, "%").replace(/\?/g, "_")}%' ESCAPE '\\'`
```